### DISCUSSION TITLE
Is there any way to retrieve the yhat estimated in the training data set?

### DISCUSSION THREAD

**Discussion by raphaccf** [pycaret/pycaret#1985] [General]
Hi there!!

It is possible to get the yhat (the estimated target variable) from predict_model(). However, this predict function only works for the test period. 

Is there any way to recover yhat from training period? 

The evaluation metrics are already calculated in several functions (compare_models; create_model; tuned_model;  blend_model), but I couldn't find any way, if it is possible, to have the yhat's. Plots of some models ( like arima, for example) also bring yhat, but not for other models, and could not find how to recovery even for such method. 
It would be important for analysis and forecast intervals.

Kind regards!

**Comment by ngupta23** [comment]
Will this work?

https://github.com/pycaret/pycaret/issues/1999#issuecomment-1004187004
